### DISCUSSION TITLE
[Query] Why use build pod instead of k8s job?

### DISCUSSION THREAD

**Discussion by leonard520** [buildpacks-community/kpack#1199] [General]
Hey! I am new to kpack. I notice that kpack will create a build job with several InitContainers to do the build staff one by one. I am out of curiosity that is k8s native job more suitable to do it? Any background? Thanks!

**Comment by leonard520** [comment]
@tomkennedy513 Would you mind taking a look? Thanks!

**Reply by tomkennedy513** [reply]
The reason we do not use kubernetes jobs is because they will create pods which is the same as we are already creating so it creates another layer of indirection that we didn't find beneficial at the time
